diff --git a/configure.ac b/configure.ac
index b8401e0..0c05839 100644
--- a/configure.ac
+++ b/configure.ac
@@ -311,20 +311,20 @@ AM_CONDITIONAL([have_libmagick_pp], [test x != "x$LIBMAGICK_PP_LIBS"])
 AS_IF([test xno != "x$with_magick"],
    [AS_CASE("x$with_magick",
      [xGraphicsMagick],
-       [AC_CHECK_PROGS([MAGICK_CONVERT], [gm])
+       [AC_CHECK_PROGS([MAGICK_CONVERT], [gm],[$MAGICK_CONVERT],[$PATH$PATH_SEPARATOR$PKG_CONFIG_SYSROOT_DIR/usr/bin$PATH_SEPARATOR])
         AS_IF([test xgm != x$MAGICK_CONVERT],
           [AC_MSG_ERROR([$with_magick requested but not found])])
         AC_DEFINE([HAVE_GRAPHICS_MAGICK], [1])
         MAGICK_CONVERT="gm convert"
        ],
      [xImageMagick],
-       [AC_CHECK_PROGS([MAGICK_CONVERT], [convert])
+       [AC_CHECK_PROGS([MAGICK_CONVERT], [convert],[$MAGICK_CONVERT],[$PATH$PATH_SEPARATOR$PKG_CONFIG_SYSROOT_DIR/usr/bin$PATH_SEPARATOR])
         AS_IF([test xconvert != x$MAGICK_CONVERT],
           [AC_MSG_ERROR([$with_magick requested but not found])])
         AC_DEFINE([HAVE_IMAGE_MAGICK], [1])
        ],
      [xyes|xcheck],
-       [AC_CHECK_PROGS([MAGICK_CONVERT], [gm convert])
+       [AC_CHECK_PROGS([MAGICK_CONVERT], [gm convert],[$MAGICK_CONVERT],[$PATH$PATH_SEPARATOR$PKG_CONFIG_SYSROOT_DIR/usr/bin$PATH_SEPARATOR])
         AS_CASE(x$MAGICK_CONVERT,
           [xgm],
             [AC_DEFINE([HAVE_GRAPHICS_MAGICK], [1])
@@ -545,9 +545,6 @@ AS_IF([test "`cd $srcdir && pwd`" != "`pwd`"],
   [AC_SUBST([AM_CPPFLAGS],
      ['-I$(top_builddir) $(BOOST_CPPFLAGS)'])])

-#  Check if we can assume whether All Warnings Are Really Errors
-AS_IF([test xyes = x$ac_cv_cxx_compiler_gnu],
-  [gcc_AWARE=-Werror])

 AC_SUBST([AM_CXXFLAGS], ["-Wall $gcc_AWARE $code_coverage_cxxflags"])
 AC_SUBST([AM_LDFLAGS], [""])

